Alternate presentation of multi-cluster exec plane names

StackScribe opened the pull request:
Signed-off-by: Meg McRoberts <meg.mcroberts@dynatrace.com>

This is an alternative presentation to https://github.com/keptn/keptn.github.io/pull/1382 , which fixes https://github.com/keptn/keptn/issues/8576 .  It makes adjusting the exec plane name a step in the process of implementing a multi-cluster environment rather than an afterthought.

The original draft had the following issues.  Giovanni responded and I believe these are fixed:
* I made up the examples without much knowledge -- please suggest better examples
* The text says that this is so the distributor can handle event subscriptions correctly, then says that the nameOverride value is available only if the integration uses the distributor.  This does not sound right.
* What happens if someone sets different values for nameOverride and K8S_DEPLOYMENT_NAME?  
* Should we add something to the Troubleshooting section to describe but it looks like if you don't handle the exec plan names correctly?

Once we get the text correct here, we need to implement these changes in the 0.19.x tree as well.

thisthat reviewed (commented):
> The text says that this is so the distributor can handle event subscriptions correctly, then says that the nameOverride value is available only if the integration uses the distributor. This does not sound right.

The distributor uses only `K8S_DEPLOYMENT_NAME`. 
Helm- and Jmeter-service configure this value based on the value of `app.kubernetes.io/name` Kubernetes label, as shown here for the jmeter-service: https://github.com/keptn/keptn/blob/master/jmeter-service/chart/templates/deployment.yaml#L105

Hence, it is up to the integration that uses the distributor to properly configure the value of `K8S_DEPLOYMENT_NAME`. In the case of helm- and jmeter-service, this can be done by setting a value for `nameOverride` in the `value.yaml` file so that the label `app.kubernetes.io/name` is configured accordingly and its value is propagated into `K8S_DEPLOYMENT_NAME`.

> What happens if someone sets different values for nameOverride and K8S_DEPLOYMENT_NAME?
`K8S_DEPLOYMENT_NAME` value wins

> Should we add something to the Troubleshooting section to describe but it looks like if you don't handle the exec plan names **correctly?**

I think it makes sense to have it👍

StackScribe commented on content/docs/install/multi-cluster/index.md:
````diff
@@ -88,6 +88,35 @@ Please find the Helm Charts here:
+
+    ```
+    distributor:
+      nameOverride: "server001-helm-server"
````
Done

StackScribe commented:
Thanks, @thisthat !  I think I incorporated your information, also a bit from the original issue, and this might be okay.  I also added a little section to "Troubleshooting" about this issue.  It isn't wonderful but perhaps good enough; we can always improve it later if necessary.

thisthat reviewed (approved):
LGTM! Thank you @StackScribe :shipit:

Lauri-Nomme commented on content/docs/install/multi-cluster/index.md:
```diff
@@ -149,19 +180,19 @@ See the configuration parameters of the supported execution plane services:
-| `distributor.stageFilter` | Sets the stage this service belongs to | `""` |
-| `distributor.serviceFilter` | Sets the service this service belongs to | `""` |
-| `distributor.projectFilter` | Sets the project this service belongs to | `""` |
+| `distributor.stageFilter` | Sets the stage to which this service | `""` |
```
seems that some words are missing in the 'Description' column

gavinservai commented:
The documentation around exec plane names definitely needs to get added to the upgrade guide. Anyone with a multi-cluster setup running through the upgrade steps will end up with a broken situation, where in the worst-case the helm-service in production picks up the events for a dev stage.
